### DISCUSSION TITLE
Question regarding filters / validation "cross fields" in Directus Cloud

### DISCUSSION THREAD

**Discussion by smontlouis** [directus/directus#14616] [Archive]
With the following:

building (one to many) departments
building (one to many) users
users (many to many) departments

A user can have departments and a building.
User can be in a building without being part of a department.

I would like it to be possible to limit the addition of departments on the user entity to departments of the same building.
Naively I tried to use validation on the departments field in the user collection, without success.

How is this possible in Directus cloud ?

**Comment by rijkvanzanten** [comment]
There's currently no way to configure this level of "connected" relationship between two fields. Please upvote https://github.com/directus/directus/discussions/10751 to show your support for this feature 🙂

**Reply by smontlouis** [reply]
I see thank you.

But let's say I create a "building admin" linked to buiding 1
Is this possible to allow him so see only users related to his building, and only departement  linked to his bulding ?

**Reply by rijkvanzanten** [reply]
Yes, that sounds like a permissions question to me. You can configure the permissions based on the current user, and relationships from the current user. If you associate a building to the user, you can then set permissions to something like `department.building _in $current_user.buildings` (simplified example)

**Reply by smontlouis** [reply]
It's working thank you.
At some points "public" departments can be seen by other buildings in the front-end app. 
I believe with this logic the users will be unable to see the other departments on the frontend app. Do you confirm ?

**Reply by rijkvanzanten** [reply]
If you configure read permissions to only allow relationally linked items, then yes. That being said, you could also configure the permissions to be `department.building _in $current_user.buildings OR department.is_public _eq true` so you allow read access for departments that are linked to the building, or departments that have a field `is_public` set to true 🙂

**Reply by smontlouis** [reply]
After looking for more info, it seems like that's what we call "multi-tenant". Great.

Thank you for those insights.

Ideally I would like the building admin to be able to see **only** the departments he can administer, while having read-only access on the front-end to public departments.

I guess this is not possible today.
Ideally it would have been nice to keep the permission to read public departments while filtering the department view for him. Is this a possible option?

A solution would be to explain to the admins that they need a "front-end" account with a user role that can see the public departments and a "back-office" account with an "is_back_office" field that can only connect to the back-office.

What do you think ?

**Reply by rijkvanzanten** [reply]
> Ideally I would like the building admin to be able to see only the departments he can administer, while having read-only access on the front-end to public departments.

You could do that! The `public` role controls what's available to users who aren't logged in, so it sounds to me like you'd configure the permissions for the public role to allow reading the departments where you have a public flag set, and set the permissions for your user roles to only allow the ones to the assigned buildings. 

> Ideally it would have been nice to keep the permission to read public departments while filtering the department view for him. Is this a possible option?

Wait so you _do_ want your admin user to be able to see those public ones as well? I'm a little confused 😅 I thought you mentioned above that you only want your "public front end" to have access to the public departments, and have your administrators only see the ones assigned to them? 🤔

**Reply by smontlouis** [reply]
Thank you for answering me.

> You could do that! The public role controls what's available to users who aren't logged in, so it sounds to me like you'd configure the permissions for the public role to allow reading the departments where you have a public flag set, and set the permissions for your user roles to only allow the ones to the assigned buildings.

Yes but if the building admin logs in with his account in the front-end app, he will not be able to see the departments with the public flag, right ?

> Wait so you do want your admin user to be able to see those public ones as well? I'm a little confused 😅 I thought you mentioned above that you only want your "public front end" to have access to the public departments, and have your administrators only see the ones assigned to them? 🤔

Yes I wabted them to see the public one **in the frontend app**, but not in the back-end app. Hence the idea of filtering the department view to keep the permissions. I believe it would make sense for them to use the same email address for the front-end and back-end apps.

**Reply by rijkvanzanten** [reply]
> Yes but if the building admin logs in with his account in the front-end app, he will not be able to see the departments with the public flag, right ?

Unless you give their role explicit permissions to see that.

> Yes I wabted them to see the public one in the frontend app, but not in the back-end app. Hence the idea of filtering the department view to keep the permissions. I believe it would make sense for them to use the same email address for the front-end and back-end apps.

Oh, and you're doing your front-end user login stuff through the same Directus user? In that case, you won't be able to handle this through Directus' permissions exclusively, as it doesn't differentiate between API use or "backend app" use. In that case, you might need to setup a middleman server that injects the additional filters when requests are made from your front-end app

**Reply by smontlouis** [reply]
Sweet ! Thank you !

**Reply by smontlouis** [reply]
@rijkvanzanten I was wondering: you seem surprised regardnig the front-end user login through the same directus auth.
Is this a common pattern ? 
How could I improve my architecture ?

**Reply by rijkvanzanten** [reply]
Logging in in both the front and backend using the same user is pretty common. That being said, I haven't heard people needing different permissions for those users based on _where_ they login from 🙂

**Reply by smontlouis** [reply]
Fair point 😅

 Hence the relevance of perhaps having a filtered view for roles.
